app/notifications/user_mentee_application/application_submission_alert.rb

```diff
@@ -0,0 +1,7 @@
+class UserMenteeApplication::ApplicationSubmissionAlert < Noticed::Base
+  deliver_by :email,
+    mailer: 'UserMenteeApplicationAlertMailer',
+    method: 'notify_for_application_submission'
```

Comment:
we've been a bit lax about this on other notifications in the code base, but I think it's important that work like mailers that can be performed asynchronously is, in fact, performed asynchronously. The only way I could confirm that it is processed via ActiveJob and properly enqueued is to add the `enqueue: true` option to the `deliver_by` method.

Reply:
hmm, I think this is a good idea, but for the sake of time, I'll probably submit a different PR for it